lawnchair/src/app/lawnchair/search/data/SearchRequest.kt

```diff
@@ -0,0 +1,206 @@
+package app.lawnchair.search.data
+
+import android.content.Context
+import android.provider.ContactsContract
+import android.provider.MediaStore
+import android.util.Log
+import app.lawnchair.search.data.suggestion.StartPageService
+import app.lawnchair.util.getMimeType
+import app.lawnchair.util.kotlinxJson
+import com.jakewharton.retrofit2.converter.kotlinx.serialization.asConverterFactory
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.ResponseBody
+import org.json.JSONArray
+import org.json.JSONObject
+import retrofit2.Response
+import retrofit2.Retrofit
+
+val retrofit: Retrofit = Retrofit.Builder()
+    .baseUrl("https://www.startpage.com/")
+    .addConverterFactory(kotlinxJson.asConverterFactory("application/json".toMediaType()))
+    .build()
+
+val startPageService: StartPageService = retrofit.create(StartPageService::class.java)
+
+suspend fun getStartPageSuggestions(query: String, max: Int): List<String> {
```

Comment:
Still think that this should be tied to the user's current search provider.
For example, a user that selected startpage as the provider should get suggestions from startpage, or a user that selected duckduckgo should get suggestions from duckduckgo.

Reply:
I see, but we don't have endpoint for other provider, we only have from startpage. and also the current dock search implementation doesnt support your suggestions. it just redirect to browser when click.

Reply:
We could make tapping the dock open the drawer search instead. That way the user could type instantly without waiting for the browser to load.

About the other providers, what happens if the user clicks a startpage suggestion while using another provider?

Reply:
Yes we can do that.

nothing, since it's not tied in the search provider in dock. I guess we can update it later on when we tied dock search bar to all appsearch